File: app/src/main/java/com/inspiredandroid/linuxcommandbibliotheca/fragments/dialogs/RateDialogFragment.java
package com.inspiredandroid.linuxcommandbibliotheca.fragments.dialogs;

import android.app.Dialog;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;

import com.inspiredandroid.linuxcommandbibliotheca.R;
import com.inspiredandroid.linuxcommandbibliotheca.misc.AppManager;
import com.inspiredandroid.linuxcommandbibliotheca.misc.Utils;

/**
 * Created by Simon Schubert
 */
public class RateDialogFragment extends DialogFragment {

    public static RateDialogFragment getInstance() {
        return new RateDialogFragment();
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        return new AlertDialog.Builder(getContext())
                .setTitle(R.string.dialog_rate_title)
                .setMessage(R.string.dialog_rate_message)
                .setPositiveButton(R.string.dialog_rate_rate_now, (dialogInterface, i) -> startAppMarketActivity())
                .setNeutralButton(R.string.dialog_rate_later, null)
                .setNegativeButton(R.string.dialog_rate_no_thanks, (dialogInterface, i) -> AppManager.disableRateDialog(getContext())).create();
    }

    private void startAppMarketActivity() {
        final String appPackageName = Utils.PACKAGE_COMMANDLIBRARY;
        try {
            startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + appPackageName)));
        } catch (android.content.ActivityNotFoundException e) {
            startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("https://play.google.com/store/apps/details?id=" + appPackageName)));
        }
    }
}

File: app/src/main/java/com/inspiredandroid/linuxcommandbibliotheca/fragments/dialogs/RateDialogFragment.kt
package com.inspiredandroid.linuxcommandbibliotheca.fragments.dialogs

import android.app.AlertDialog
import android.app.Dialog
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import com.inspiredandroid.linuxcommandbibliotheca.R
import com.inspiredandroid.linuxcommandbibliotheca.misc.AppManager
import com.inspiredandroid.linuxcommandbibliotheca.misc.Constants
import com.inspiredandroid.linuxcommandbibliotheca.misc.showPlayStoreActivity

/* Copyright 2019 Simon Schubert
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/
class RateDialogFragment : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return AlertDialog.Builder(requireContext())
                .setTitle(R.string.dialog_rate_title)
                .setMessage(R.string.dialog_rate_message)
                .setPositiveButton(R.string.dialog_rate_rate_now) { _, _ -> context?.showPlayStoreActivity(Constants.PACKAGE_COMMANDLIBRARY) }
                .setNeutralButton(R.string.dialog_rate_later, null)
                .setNegativeButton(R.string.dialog_rate_no_thanks) { _, _ -> AppManager.disableRateDialog(context) }.create()
    }

    companion object {

        val instance: RateDialogFragment
            get() = RateDialogFragment()
    }
}
